GVRf/Framework/framework/src/main/jni/engine/renderer/renderer.cpp

```diff
@@ -348,14 +348,17 @@ void addRenderData(RenderData *render_data) {
 void Renderer::occlusion_cull(Scene* scene,
         std::vector<SceneObject*>& scene_objects, ShaderManager *shader_manager,
         glm::mat4 vp_matrix) {
-
+    scene->lockColliders();
```

Comment:
I could change it to not lock if the allColliders list is empty. I am not sure this is 100% safe. It assumes colliders cannot be added during rendering. We don’t lock the scene graph during cull. Instead we call SceneObject::children which looks like:

    std::vector<SceneObject*> children() {
        std::lock_guard < std::mutex > lock(children_mutex_);
        return std::vector<SceneObject*>(children_);
    }
Apparently we make a copy of the child list as we do culling. This can’t be efficient either. I think the whole cull thing needs some rework.

From: rahul [mailto:notifications@github.com]
Sent: Monday, June 27, 2016 11:39 AM
To: Samsung/GearVRf <GearVRf@noreply.github.com>
Cc: Nola Donato <nola.donato@samsung.com>; Author <author@noreply.github.com>
Subject: Re: [Samsung/GearVRf] Picking Reform (#621)


In GVRf/Framework/framework/src/main/jni/engine/renderer/renderer.cpp<https://github.com/Samsung/GearVRf/pull/621#discussion_r68631997>:

> @@ -348,14 +348,17 @@ void addRenderData(RenderData *render_data) {

>  void Renderer::occlusion_cull(Scene* scene,

>          std::vector<SceneObject*>& scene_objects, ShaderManager *shader_manager,

>          glm::mat4 vp_matrix) {

> -

> +    scene->lockColliders();

Is it okay to have the overhead of performing a lock/unlock and pick check if there are no colliders or pickers in the scene?

Just curious since this is the renderer's code.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/Samsung/GearVRf/pull/621/files/80809f64dabace008becc07651f4c223ff73b816..7d4d25bac296f8c6dae91028a9441ba0745816a0#r68631997>, or mute the thread<https://github.com/notifications/unsubscribe/AQw1S5aTFRSRzzCAzeNuj3RtbpVim4Ieks5qQBjXgaJpZM4I6MsB>.